modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -1,197 +1,138 @@
-// and on pressing the gear icon, a file will be created
-// If one file is present, the chosenFile is set to that file, and the action is hidden, it is set as default
-// If two or more files are present, the chosenFile is set to the first file, and the action is shown
-// Startup -> MirrordConfigIndex -> (MirrordConfigWatcher -> Event -> Query Index -> Update Configs) -> MirrordConfigDropDown
-//                                    \<--------------------------------------------------------/
 class MirrordConfigDropDown : ComboBoxAction() {
+    private class EnableMirrordAction(val enabled: Boolean) : AnAction(if (enabled) "Enabled" else "Disabled") {
+        override fun actionPerformed(e: AnActionEvent) {
+            val service = e.project?.service<MirrordProjectService>() ?: return
+            service.enabled = !enabled
+        }
+    }
 
-    private lateinit var configFiles: HashSet<String>
+    private class ShowActiveConfigAction(val config: VirtualFile) : AnAction("Active Config: ${config.presentableUrl}") {
+        override fun actionPerformed(e: AnActionEvent) {
+            val service = e.project?.service<MirrordProjectService>() ?: return
+            FileEditorManager.getInstance(service.project).openFile(config, true)
+        }
+    }
 
-    @Volatile
-    private var blockQueries: Boolean = false
+    private class SelectActiveConfigAction : AnAction("Select Active Config") {
+        override fun actionPerformed(e: AnActionEvent) {
+            val service = e.project?.service<MirrordProjectService>() ?: return
 
-    override fun getActionUpdateThread() = ActionUpdateThread.BGT
+            val fileManager = VirtualFileManager.getInstance()
+            val projectLocator = ProjectLocator.getInstance()
+            val configs = FileBasedIndex
+                .getInstance()
+                .getAllKeys(MirrordConfigIndex.key, service.project)
+                .mapNotNull { fileManager.findFileByUrl(it) }
+                .filter { projectLocator.getProjectsForFile(it).contains(service.project) }
+                .associateBy { it.presentableUrl }
 
-    // this function is called on click of the dropdown, here we map configFiles -> AnAction
-    override fun createPopupActionGroup(button: JComponent, dataContext: DataContext): DefaultActionGroup {
-        val project = dataContext.getData(CommonDataKeys.PROJECT) ?: throw Error("couldn't resolve project")
-        val actions = configFiles.map { configPath ->
-            object : AnAction(getReadablePath(configPath, project)) {
-                override fun actionPerformed(e: AnActionEvent) {
-                    chosenFile = configPath
-                }
-            }
-        }
-        return DefaultActionGroup().apply {
-            addAll(actions)
-        }
-    }
+            val selection = MirrordConfigDialog(
+                "Change mirrord active configuration",
+                configs.keys.toList().sorted()
+            ).show() ?: return
```

Comment:
for shorter paths this title doesnt show up properly, maybe consider shortening it?

Reply:
I made the dialog wider to fit the title

Reply:
awesome!